src/BrickbattleWeapons/MainModule/ServerModules/slices/BrickbattleWeaponSlice.lua

```diff
@@ -0,0 +1,42 @@
+--!strict
+local Reflex = require("Packages/reflex")
+local Immut = require("Packages/immut")
+local Table = require("src/BrickbattleWeapons/MainModule/ServerModules/slices/BrickbattleWeaponTables")
+
+export type BrickbattleWeaponStateActions = {
+    setStatState: (playerName: string, numb: number, stat: string) -> (),
+    removeStatState: (playerName: string, stat: string) -> (),
+}
+
+type Data = {
+    [string]: Table.BrickbattlePlayerTable
+}
+local initialState: Data = {}
+local BrickbattlePlayerStateSlice = Reflex.createProducer(initialState, {
+    setPlayerWeaponState = function(state, playerName: string,weapon: string, Data: any)
+		return Immut.produce(state, function(draft) 
+            local playerData = draft[playerName]
+            if playerData == nil then return end
+            playerData[weapon] = Data
+            return draft
+        end)
+    end,
+    setStatWeaponState = function(state, playerName: string, weapon: string, weaponCount: number, Amount: any)
+		return Immut.produce(state, function(draft) 
+            local playerData = draft[playerName]
+            if playerData == nil then return end
+            playerData[weapon][weaponCount] = Amount
+            return draft
+        end)
+    end,
+    removeStatState = function(state, playerName: string, weapon: string, weaponCount: number)
+		return Immut.produce(state, function(draft) 
+            local playerData = draft[playerName]
+            if playerData == nil then return end
```

Comment:
shouldn't we error here if player data that we want to remove is already nil?

Reply:
I don't think it will error, just won't upd the state

Reply:
let's add an error log in the case that playerData is nil